pkg/controller/lifecycle/actuator.go

```diff
@@ -695,6 +708,31 @@ func getShootResources(webhookCaBundle []byte, extensionNamespace, shootAccessSe
 	return shootResources, nil
 }
 
+func getClientKeys(ctx context.Context, client client.Client, resources []v1beta1.NamedResourceReference, resourceName, namespace string) ([]byte, error) {
+    ref := v1beta1helper.GetResourceByName(resources, resourceName)
+    if ref == nil || ref.ResourceRef.Kind != "Secret" {
```

Comment:
I will propose the split the condition and have dedicated error messages for both cases, one is that the configured `ResourceName` is missing in the `shoot.spec.resources`, the other is that it not of the expected kind `Secret`.